### PR TITLE
Support complete builds from source of pyoorb

### PR DESCRIPTION
**Pull request by spenczar**
This adds support for building Pyoorb entirely from source, as long as the caller has gfortran installed. The source code can be bundled up into a tarball and put on PyPI, which will allow for users to build pyoorb on their own. This should help for ARM Macbooks, cutting-edge Python releases, etc.

This change has the following components:
1. A new `MANIFEST.in` file (docs [here](https://packaging.python.org/en/latest/guides/using-manifest-in/)) which lists all the files that ought to be included in the source tarball./
2. Changes to move source code out of the `./build` directory - there were two previously. It turns out that Python packaging tools refuse to ever include the `./build` directory in source tarballs - that name is special, and is used when Python is installing a module. So the top-level `Makefile` now creates the ./build directory at build time, and moves the old source in there from `./build-tools`.
3. The `setup.py` build script now runs the `configure` script to set things up before the build begins, alerting it to the location of f2py and python and the fortran compiiler.

---

This is working locally for me when running `pipx run build --sdist`, which is what the github actions workflow does. I can `pip install` the resulting tarball.
```console
pipx run build --sdist
# much output...
pip install -v ./dist/pyoorb-1.2.0.post45+0b9a95b.dirty.tar.gz
# yet more output, including compilation...
```

We'd certainly want to cut a prerelease to see if this is actually working end-to-end though.
```

### CODE REVIEW

**Comment by spenczar** [comment]
I believe the most recent test failure is spurious. Error code 28 from curl means a timeout when downloading ephemeris data from JPL.

**Comment by Siltala** [comment]
Thanks for the PR!

Agreed on the test failure, I restarted the test and it finished successfully, which confirms your explanation.

I'll take a closer look soon, but the one comment I have is that I appreciate having a clean commit history and minimizing needless commits. To that end, I think it would make sense to squash [94374d5](https://github.com/oorb/oorb/pull/177/commits/94374d55aedb67a34b1bd5a06a90692f3ab79480) and [3866a77](https://github.com/oorb/oorb/pull/177/commits/3866a7756bea4d9c17dffb64ddd9b725369664cc) together, and same for [e13c63a](https://github.com/oorb/oorb/pull/177/commits/e13c63ace2d0ac5d6e02836df11ca767cb61de16) and [940202f](https://github.com/oorb/oorb/pull/177/commits/940202f7685a32baf3bdb2e7cc64e34bf9bc08c1) since they do the same thing / one supersedes the other.

**Comment by spenczar** [comment]
Of course! Will do. I like to squash just once when we agree on the content of the changes, just to avoid needless force-pushes.

**Comment by Siltala** [comment]
Aside from the one unnecessary whitespace above and squashing the commits, this looks fine and I'll merge once that's done.

Right now I feel like I would like to get #171 done also (waiting for a review from Mikael for a while now) and then do a new release (with a pre-release at first to test this with, as you suggest) to have a new and useful feature as part of it beyond build system updates. But if that ends up taking too long, we'll see :)

File: Makefile [hunk 1]
```diff
@@ -60,18 +60,23 @@ ifeq ("$(PREFIX)", "")
...
+	mkdir -p build
+	cp build-tools/make.depends build/make.depends
+	cp build-tools/Makefile build/Makefile
 
 ephem:
 	@ $(MAKE) -C data/JPL_ephemeris
 
 # Forward everything we don't recognize to the makefile in build/
-%:
+%:
```
[the remark this answers is not in the document]
Comment (spenczar) [reply Makefile:75]: Oops! yes.

**Pull request merged**